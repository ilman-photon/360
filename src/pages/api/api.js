import axios from "axios";
export class Api {
  client;
  constructor(url) {
    this.client = axios.create({
      headers: {
        "Content-Type": "application/json",
        Accept: "application/json",
      },
      timeout: 10000,
    });
  }
}
