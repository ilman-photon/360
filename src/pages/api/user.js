import { faker } from "@faker-js/faker";

export default function handler(req, res) {
  res.status(200).json({
    firstName: faker.name.firstName(),
    lastName: faker.name.lastName(),
  });
}
