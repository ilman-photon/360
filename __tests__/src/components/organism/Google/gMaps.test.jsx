import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import React from "react";
import GMaps from "../../../../../src/components/organisms/Google/Maps/gMaps";
import { CircularProgress } from "@mui/material";
import { Marker, useLoadScript } from "@react-google-maps/api";
import ShallowRenderer from 'react-shallow-renderer';

const originalEnv = process.env;

beforeEach(() => {
  jest.resetModules();
  process.env = {
    ...originalEnv,
    NODE_ENV: 'test',
  };
});

afterEach(() => {
  process.env = originalEnv;
});

jest.mock("react", () => {
  const originReact = jest.requireActual("react");
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef,
  };
});

jest.mock("@react-google-maps/api", () => ({
  useLoadScript: () => ({
    isLoaded: true,
    loadError: null
  }),
  GoogleMap: () => <div></div>,
  Marker: () => <Marker />
}));

describe("GMaps", () => {
  const renderer = new ShallowRenderer();
  const { isLoaded } = useLoadScript({
    googleMapsApiKey: process.env.GOOGLE_API_KEY
  });
  let container;
  beforeEach(() => {
    container = renderer.render(
      isLoaded ? <GMaps apiKey={process.env.GOOGLE_API_KEY} /> : <CircularProgress/>
    );
  });
  it("GMaps render", () => {
    expect(container).toMatchSnapshot();
  });
});
