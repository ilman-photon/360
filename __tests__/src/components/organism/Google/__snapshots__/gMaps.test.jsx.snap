// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GMaps GMaps render 1`] = `
<GoogleMap
  center={
    Object {
      "lat": 41.881832,
      "lng": -87.623177,
    }
  }
  mapContainerStyle={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
  onLoad={[Function]}
  onUnmount={[Function]}
  zoom={4}
>
  <UNDEFINED
    onClick={[Function]}
    position={
      Object {
        "lat": 41.881832,
        "lng": -87.623177,
      }
    }
  />
  <UNDEFINED
    onClick={[Function]}
    position={
      Object {
        "lat": 39.739235,
        "lng": -104.99025,
      }
    }
  />
  <UNDEFINED
    onClick={[Function]}
    position={
      Object {
        "lat": 34.052235,
        "lng": -118.243683,
      }
    }
  />
  <UNDEFINED
    onClick={[Function]}
    position={
      Object {
        "lat": 40.712776,
        "lng": -74.005974,
      }
    }
  />
</GoogleMap>
`;
